admin/RequestSolrCloudBackups

```diff
@@ -0,0 +1,307 @@
+                $backup_name,
+                $solr_response->{error},
+            );
+            _flush_running_requests_status();
+            exit 70; # EX_SOFTWARE
+        }
+        $backup_requests{$request_id}{status} = 'running';
+    }
+}
+
+sub wait_backup_complete
+{
+    while (
+        sum
+        map { $_ => 1 }
+        grep { $backup_requests{$_}{status} eq 'running' }
+        keys %backup_requests
+    )
+    {
+        my $pause = '60s';
+        _debug "Sleeping $pause";
+        sleep $pause;
+        foreach my $request_id (
+            sort
+            grep { $backup_requests{$_}{status} eq 'running' }
+            keys %backup_requests
+        )
+        {
+            my $solr_response = _query_collections_api('REQUESTSTATUS', {
+                    requestid => $request_id,
+                }
+            );
+            _debug (
+                "Received: %s",
+                encode_json($solr_response),
+            );
+            if (exists $solr_response->{error}) {
+                _error(
+                    "Failed on backup request '%s' of collection '%s' as '%s': " .
+                    "Error '%s'",
+                    $request_id,
+                    $backup_requests{$request_id}{collection},
+                    $backup_requests{$request_id}{name},
+                    $solr_response->{error},
+                );
+                _flush_running_requests_status();
+                exit 70; # EX_SOFTWARE
+            }
+            elsif ($solr_response->{status}->{state} eq 'completed')
+            {
+                $backup_requests{$request_id}{status} = 'completed';
+                _query_collections_api('DELETESTATUS', {
+                        requestid => $request_id,
+                    }
+                );
+            }
+            elsif ($solr_response->{status}->{state} ne 'running')
+            {
+                _error(
+                    "Failed on backup request '%s' of collection '%s' as '%s': " .
+                    "Unhandled state '%s'",
+                    $request_id,
+                    $backup_requests{$request_id}{collection},
+                    $backup_requests{$request_id}{name},
+                    $solr_response->{status}->{state},
+                );
+                _flush_running_requests_status();
+                exit 70; # EX_SOFTWARE
+            }
+        }
+    }
+}
+
+sub _flush_running_requests_status
+{
+    foreach my $request_id (
+        sort
+        grep { $backup_requests{$_}{status} eq 'running' }
+        keys %backup_requests
+    )
+    {
+        _query_collections_api('DELETESTATUS', {
+                requestid => $request_id,
+            }
+        );
+    }
+}
+
+sub _query_collections_api
+{
+    my ($action, $parameters) = @_;
+
+    my $url = $COLLECTIONS_API . '?action=' . $action;
+    foreach my $key (keys %{$parameters})
+    {
+        $url .= '&' . $key . '=' . $parameters->{$key};
+    }
```

Comment:
`URI::QueryParam` might be nifty here, e.g. `$u->query_param(action => $action, %{$parameters});`. (You should be able to pass a `URI` object directly to `$c->lwp->get`.)

Reply:
> `URI::QueryParam` might be nifty here

Indeed, that should prevent bad surprises in the future if that code is used with parameter values that need to be escaped. Used in https://github.com/metabrainz/musicbrainz-server/pull/1407/commits/4ec2f862faf100fd12d64c962c3b705e3c3e66ef

> e.g. `$u->query_param(action => $action, %{$parameters});`

_Do NOT do this_ according to https://metacpan.org/pod/URI::QueryParam

Reply:
Lol good call. That's what happens when you skim the examples and don't see the giant NOT in front of them. :D